sys/newlib/syscalls.c

```diff
@@ -186,7 +186,7 @@ int _open_r(struct _reent *r, const char *name, int flags, int mode)
  *
  * @return      TODO
  */
-int _read_r(struct _reent *r, int fd, void *buffer, unsigned int count)
+_ssize_t _read_r(struct _reent *r, int fd, void *buffer, size_t count)
```

Comment:
Has nothing to do with this PR, but what's the difference between _ssize_t and ssize_t?